README-dnssec.md

```diff
@@ -0,0 +1,296 @@
+# DNSSEC setup for dnsseed
+
+##### Table of Contents  
+[Requirements](#Requirements)  
+[Software needed](#Software)  
+[Setup](#Setup)  
+[Testing](#Testing)  
+[Example](#Example)  
+[Improvements](#Improvements)  
+[Links](#Links)  
+
+
+<a name="Requirements"/>
+
+## Requirements
+* run only this service on the host (security precaution)
+
+<a name="Software"/>
+
+## Software needed
+
+* tor (if you like to seed tor nodes)
+* bind9
+* dig (debian package bind9-dnsutils)
+* nsupdate (debian package bind9-dnsutils)
+* bitcoin-seeder
+* ufw
+* bash
+
+```
+apt update
+apt install bind9 bind9-dnsutils bind9utils ufw tor
+```
+
+<a name="Setup"/>
+
+## Setup
+
+### ufw
+Configure and enable firewall
+```
+ufw allow 22/tcp
+ufw allow 53/udp
+ufw allow 53/tcp
+ufw enable
+ufw status
+```
+
+### Configuring Bind9
+Replace `example.com` with your domain.
+
+* add or change in /etc/bind/named.conf.options 
+```
+        querylog no;
+	allow-transfer { none; };
+	dnssec-enable yes;
+```
+
+* add to /etc/bind/named.conf.local 
+```
+zone "dnsseed.example.com" {
+      type master;
+      file "/var/lib/bind/db.dnsseed.example.com";
+      key-directory "/var/lib/bind";
+      allow-update {localhost;};
+      auto-dnssec maintain;
+      inline-signing yes;
+};
+```
+
+* Generate new file /var/lib/bind/db.example.com
+```
+$ORIGIN .
+$TTL 3600       ; 1 hour
+dnsseed.example.com.        IN SOA  dnsseed-host.example.com. contact-email.example.com. (
+                                2000000001 ; serial
+                                3600       ; refresh (1 hour)
+                                600        ; retry (10 minutes)
+                                86400      ; expire (1 day)
+                                600        ; minimum (10 minutes)
+                                )
+                        NS      dnsseed-host.example.com.
+dummy                   A       127.0.0.1
+```
+
+* Check config
+`named-checkconf`
+`named-checkzone dnsseed.example.com /var/lib/bind/db.dnsseed.example.com`
+
+* Restart Bind9
+`service bind9 restart`
+
+* Generate DNSSEC keys
+```
+cd /var/lib/bind
+dnssec-keygen -r /dev/urandom -a ECDSAP256SHA256 dnsseed.example.com`
+dnssec-keygen -r /dev/urandom -a ECDSAP256SHA256 -b 2048 -fKSK -n ZONE dnsseed.example.com
+chown bind: Kdnsseed.example.com*.key
+```
+
+* Add keys to zone file
+```
+cd /var/lib/bind
+for key in `ls Kdnsseed.example.com*.key`
+do
+echo "\$INCLUDE $key">> db.example.com
+done
+```
+
+* Sign zone
+```
+dnssec-signzone -A -3 $(head -c 1000 /dev/random | sha1sum | cut -b 1-16) -N INCREMENT -o example.com -t db.example.com
```

Comment:
It would be nice to document why the randomness is needed here. You could assign it to a variable and have it on a separate line. unset it after, if you like. I think it would be more readable, and the binding name would explain what you're trying to do.

Reply:
The salt is intended to deter pre-computed dictionary attacks.

I like to keep the howto short. I don't like to explain DNSSEC here.